systems/robotInterfaces/QPLocomotionPlan.m

```diff
@@ -678,14 +684,32 @@ function plot_traj_foh(traj, color)
       end
       q0 = qtraj.eval(qtraj.tspan(1));
       x0 = [q0; zeros(biped.getNumVelocities(), 1)];
-      obj = QPLocomotionPlan.from_standing_state(x0, biped);
+      obj = QPLocomotionPlan(biped);
+      obj.x0 = x0;
       obj.qtraj = qtraj;
       obj.duration = obj.qtraj.tspan(end) - obj.qtraj.tspan(1);
-      obj.support_times = [obj.qtraj.tspan(1); inf];
-
-      if isfield(options,'supports') && isfield(options,'support_times')
-        obj.supports = options.supports;
-        obj.support_times = options.support_times;
+      obj.support_times = options.support_times;
+      obj.supports = options.supports;
+      obj.is_quasistatic = options.is_quasistatic;
+      obj.gain_set = options.gain_set;
+
+      obj = obj.setCOMTraj();
+      if ~obj.is_quasistatic
+        support_state = obj.supports(1);
+        for j = 1:length(obj.supports)
+          if any(obj.supports(j).bodies ~= support_state.bodies)
+            error('We don''t know how to construct non-quasistatic manipulation plans in which the active supports change. Try going slowly and setting options.is_quasistatic to true, or just make a walking plan.')
+          end
+        end
+        active_contact_pts = zeros(3,0);
+        kinsol = obj.robot.doKinematics(q0);
+        for j = 1:length(support_state.bodies)
+          active_contact_pts = [active_contact_pts, obj.robot.forwardKin(kinsol, support_state.bodies(j), support_state.contact_pts{j}, 0)];
+        end
+        comgoal = mean(active_contact_pts(1:2,:), 2);
+        obj.zmptraj = comgoal;
+        obj.zmp_final = obj.zmptraj;
```

Comment:
also, would it make sense to interpolate from the current zmp to here, instead of transitioning immediately?